### ISSUE TITLE
Drive Loader Sound Emulation

### ISSUE THREAD

**Issue by reteo** [atari800/atari800#110]
More of a wishlist, really... I enjoy using Atari800 to play the games I used to when I was a kid, but I kinda miss listening to the beeping pattern when loading software from floppy.  I'd love to see that find its way into the emulator at some point.

**Comment by mikrosk** [comment]
Actually, it is there. However under one condition (which I have of course forgotten :-)) it is muted (maybe with stereo enabled?)

Also, don't forget that some games/demos intentionally disable this sound even on real hardware.

**Comment by TheRobotFactory** [comment]
IIRC the loading sounds are enabled when SIO Patch fast loading is disabled.

**Comment by reteo** [comment]
Ah, I see what you mean.  However, it doesn't really have the same sound as the original Atari 8-bit computers' loading sequences.  My initial assessment is that the loading process is still moving at a significantly higher rate than was done on the old Atari hardware, which is why the longer beeps never happen, and there are no pauses where a new file access is being performed.

As for the muting, I remember that, too... it sounded like the volume was turned way down, and the pitch of the sound went way up, but even then, it was never entirely silent.

[What I was hoping for](https://www.youtube.com/watch?v=tzIksmZNjME)
[What I received](https://youtu.be/0ZECuLelfHY)

Like I said, however, it's just a wishlist for a pique of nostalgia.  Or do I need to change another setting somewhere?

**Comment by dmsc** [comment]
Hi!

There are many factors that alter the SIO sounds in the Atari:
- The harsh sounds are produced by the actual bits being transferred, and depends on the exact bit pattern, so if you are reading a sector with $55 or $AA bytes it will sound at a high pitch of 9600Hz (half the baud rate), and $33 or $CC will produce a pitch of 4800Hz, etc.
- The "tone" is produced because the audio output is toggled between each received byte, so at 19200 bauds (the SIO default frequency), it produces a 960Hz tone. As this tone depends of the speed at which the disk drive sends each byte, different drive models produce slightly different notes on load - the 810 drive produces 943.4Hz and the 1050 is a lower tone at 910.8Hz. This tone is only produced on read operations, so writes don't have this sound.
- The "beeps" are full sector transfers, and the spacing depends on the exact position of the reading head in the disk drive. As the disk is rotating, when receiving the command to read a sector the drive starts scanning the disk to find the requested sector. So, the spacing of the beeps depends on how the disk was formatted (the sector interleave) and the speed at which the computer is reading the data. On modern SIO2PC disk drive emulators, you don't have a rotating disk, so all the beeps are evenly spaced.
- And at last, when the disk drive needs to change the track, you have a long pause to reach the required track before starting to read a sector. This produces the long gaps after 18 sectors (in the 810) or after 26 sectors (in the 1050 enhanced format). The drive will produce a loud sound when moving the head.
- And the last one, because real-life circuits are not perfect, even when you mute the sounds in a real Atari the SIO data line "bleeds" to the audio-in line, so you always hear some of the SIO data sounds.

The Atari800 emulator only produces the first sound (the serial data sound), adding the second one should not be difficult.
To add the spacing the emulator would need to emulate the full rotating disk, at the correct RPM, to calculate when the next sector is available to read. This is not easy.

**Comment by reteo** [comment]
Hence why it's just a wishlist item, and not something urgent.  I figured I'd make the request in case someone can figure something out, so I could indulge my nostalgia. ;)

As for the irregular timing, I figured it was likely based on the mechanics of the drive; when I was loading stuff, that became my guaranteed method of telling when something was wrong, as I started to memorize the sounds each disk made when loading.  After all, if any of the beeps sounded different, or changed their timing, then something was wrong.

When the drive went bad, I knew it immediately.  At that point, the parents decided to go C128, so no more Atari. :(

**Comment by bebing2** [comment]
Please address this issue if someone can! It is one of my favorite things that triggers full nostalgia! Here is the write up I prepared before seeing it was already an open issue.

I believe at one point this was working correctly, but was unable to pin-point when the change occurred. On real hardware, SIO disk reads are "beeps" and SIO disk writes are "clicks" for lack of more descriptive words. There is also well-known "buzzing" type noises when real hardware is waiting for responses(acks or nacks).

When it was working correctly, "read" beeps sounded like beeps and "write" clicks sounded like clicks, and "wait" buzzing sounded like real hardware.

Now, "read" beeps sound more or just like "write" clicks. "Write" clicks sound about right. The "wait" buzzing sounds very tinny and thin, not thick and full like on real hardware.

I suppose many people would not notice because they use the fast SIO patch, but I like to use it as if it is a stock Atari machine. I can maybe help with a patch with a bit of guidance, but it seems tied to what Pokey is doing with the data lines which I'm not to knowledgeable in.

**Comment by mikrosk** [comment]
Just for the record, in the whole thread: https://sourceforge.net/p/atari800/mailman/atari800-users/thread/CAN5rfbQt0AoD-bnc2U%3DWu6uo9k23qv7Y_dovZRd5psWe9EZeEw%40mail.gmail.com/#msg58763436 neither of us were able to confirm whether, when and how exactly the sound used to better.

From my testing between 1.3.6 - 5.2.0 builds (yes, I built each one of them!) with High Fidelity POKEY ON (where applicable), SIO Sound ON (where applicable) and SIO Patch OFF I couldn't find any case where an older version sounded actually better than a newer one. I also tried the volume only sound (instead of synchronized sound), sounded like sh*t.

So unless proven otherwise, I'd say this is a feature request, not regression.

**Comment by bebing2** [comment]
I would say it's a requirement for a program that aims to emulate the original hardware, I believe the cassette loading sounds are accurate. That's just my humble opinion, not a specific classification for an issue reporting website.

**Comment by mikrosk** [comment]
I completely agree, hearing it in Altirra made me immediately scream "Yes, this is how it is supposed to sound!" :)

**Comment by bebing2** [comment]
Awesome! Just my opinion, but I would keep the SIO noises that were generated by Pokey and heard through the speaker via the audio out of the computer as a separate issue than manufacturing noises that mimic the physical noises of the disk drive. I am going to start poking through the code, but I won't attempt to create physical disk grinding noises or anything like that, just what Pokey sent to the audio channel. Obviously just my preferences, everyone is different.

**Comment by bebing2** [comment]
Still looking into this. For my build it doesn't look like the code goes through SERIO_SOUND at all, but through CONSOLE_SOUND. Decided to learn more starting from the original hardware and documentation about the original hardware. Found this thread on atariage quickly in my search that has mp3s and some information on why different drives and speed sound different:

https://forums.atariage.com/topic/190073-looking-for-mp3-of-atari-booting-sound-anyone-have/#comment-2404273

**Comment by bebing2** [comment]
The author of Altirra gave quite a detailed response to my post at AtariAge:

https://forums.atariage.com/topic/190073-looking-for-mp3-of-atari-booting-sound-anyone-have/?do=findComment&comment=5455202

Not sure if I will have the time to look at this much further.
